internal/server/run_test.go

```diff
@@ -0,0 +1,224 @@
+package server_test
+
+// Using separate _test package to avoid circular dependency with import of "tcp-load-balancer/test" package.
+
+import (
+	"net"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"tcp-load-balancer/internal/server"
+	"tcp-load-balancer/internal/upstream"
+	"tcp-load-balancer/test"
+
+	"github.com/google/uuid"
+)
+
+func Test_ForwardData(t *testing.T) {
+	uniquePayload := uuid.New().String()
+	tests := []struct {
+		name                 string
+		hostConn             net.Conn
+		clientConnectionPipe func() (net.Conn, net.Conn)
+		payload              string
+		wantErr              bool
+	}{
+		{
+			name: "Data is forwarded to host, and response includes original payload",
+			hostConn: func() net.Conn {
+				h, err := test.InitializeHost("tcp", ":0")
+				if err != nil {
+					t.Error(err)
```

Comment:
Any errors here will prevent the rest of the test from succeeding. Might as well abort early. Same for other tests case setup.
```suggestion
					t.Fatal(err)
```